src/hotspot/share/cds/heapShared.cpp

```diff
@@ -286,6 +286,7 @@ bool HeapShared::archive_object(oop obj) {
     obj->identity_hash();
     CachedOopInfo info = make_cached_oop_info(obj);
     archived_object_cache()->put(obj, info);
+    archived_object_cache()->maybe_grow();
```

Comment:
Is it possible to use `put_if_absent()` and only calls `maybe_grow()` if an entry has been created? Similar to what is being done in `HeapShared::add_to_dumped_interned_strings()`.

Reply:
The entry is guaranteed to not exist. I changed the call to `put_when_absent()`, which asserts that no current entry exists. (Yes, the naming of `put_when_absent()` is confusing w.r.t. `put_if_absent()`).